analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTests.cs

```diff
@@ -103,49 +95,38 @@ public void WhenHasUnexpectedProjectType_FallsBackToProduct()
 
             sut.FilesToAnalyzePath.Should().Be(@"c:\foo\bar\.sonarqube\conf\0\FilesToAnalyze.txt");
         }
 
-        [DataTestMethod]
-        [DataRow(null)]
-        [DataRow("/foo/bar/do-not-exit")]
-        [DataRow("/foo/bar/x.xml")]
-        public void WhenNoFileFound_ReturnsNull(string folder)
-        {
-            var options = TestHelper.CreateOptions(folder);
-
-            var sut = new ProjectConfigReader(options);
-
-            sut.AnalysisConfigPath.Should().BeNull();
-            sut.ProjectPath.Should().BeNull();
-            sut.FilesToAnalyzePath.Should().BeNull();
-            sut.OutPath.Should().BeNull();
-            sut.ProjectType.Should().Be(ProjectType.Product);
-            sut.TargetFramework.Should().BeNull();
-        }
-
         [DataTestMethod]
         [DataRow("Invalid_DifferentClassName")]
         [DataRow("Invalid_DifferentNamespace")]
         [DataRow("Invalid_Xml")]
         public void WhenInvalid_FilesToReturnPath_ThrowsException(string folder)
         {
-            var options = TestHelper.CreateOptions($"ResourceTests\\SonarProjectConfig\\{folder}\\SonarProjectConfig.xml");
+            Action a = () => CreateProjectConfigReader($"ResourceTests\\SonarProjectConfig\\{folder}\\SonarProjectConfig.xml");
 
-            var sut = new ProjectConfigReader(options);
-            sut.Invoking(x => x.AnalysisConfigPath).Should().Throw<InvalidOperationException>().WithMessage("File SonarProjectConfig.xml could not be parsed.");
+            a.Should().Throw<InvalidOperationException>().WithMessage("File LogNameFor-SonarProjectConfig.xml could not be parsed.");
         }
 
         [TestMethod]
-        public void WhenFileIsMissing_ThrowException()
+        public void FilesToAnalyze_LoadsFileFromConfig()
         {
-            var options = TestHelper.CreateOptions("ThisPathDoesNotExist\\SonarProjectConfig.xml");
-            var sut = new ProjectConfigReader(options);
-            sut.Invoking(x => x.AnalysisConfigPath).Should().Throw<InvalidOperationException>().WithMessage("File SonarProjectConfig.xml has been added as an AdditionalFile but does not exist.");
+            var config = SourceText.From(@"
+<SonarProjectConfig xmlns=""http://www.sonarsource.com/msbuild/analyzer/2021/1"">
+    <FilesToAnalyzePath>ResourceTests\FilesToAnalyze\FilesToAnalyze.txt</FilesToAnalyzePath>
+</SonarProjectConfig>");
+
+            var sut = new ProjectConfigReader(config, null);
+            var files = sut.FilesToAnalyze.FindFiles("web.config");
+
+            files.Should().HaveCount(2);
+            files.Should().BeEquivalentTo(new[] { @"C:\Projects/DummyProj/wEB.config", @"C:\Projects/DummyProj/Views\Web.confiG" });
         }
+
+        private ProjectConfigReader CreateProjectConfigReader(string relativePath) =>
+            new ProjectConfigReader(SourceText.From(File.ReadAllText(relativePath)), "LogNameFor-SonarProjectConfig.xml");
```

Comment:
New expression can be simplified

```suggestion
            new (SourceText.From(File.ReadAllText(relativePath)), "LogNameFor-SonarProjectConfig.xml");
```